contracts/Registry/TokenCuratedRegistry.sol

```diff
@@ -1,59 +1,106 @@
 pragma solidity ^0.4.24;
 
 import './StakedRegistry.sol';
-import '../Challenge/ChallengeFactory.sol';
+import '../Challenge/IChallengeFactory.sol';
+import '../Challenge/IChallenge.sol';
 import 'openzeppelin-solidity/contracts/token/ERC20/ERC20.sol';
 
 contract TokenCuratedRegistry is StakedRegistry {
   uint public applicationPeriod;
-  ChallengeFactory public challengeFactory;
+  IChallengeFactory public challengeFactory;
```

Comment:
Should there me a function in this contract that uses this factory to create challenges?

Reply:
woops, forgot a `challenge` function!